--- uqwk-2.16/configure.in	Tue Apr 20 15:12:57 1999
+++ uqwk-2.16-p/configure.in	Tue Sep  5 13:24:32 2000
@@ -33,6 +33,7 @@
 dnl first, the socket functions
 SOCKET_FOUND=no
 SOCKLIBS=
+AC_CHECK_FUNC(socket, [SOCKET_FOUND=yes], [
 AC_CHECK_LIB(socket, socket,[SOCKET_FOUND=yes;SOCKLIBS="-lsocket"])
 if test ${SOCKET_FOUND} = no; then
   AC_CHECK_LIB(bsd, socket,[SOCKET_FOUND=yes;SOCKLIBS="-lbsd";DEFS="$DEFS -D_BSD=43"])
@@ -40,13 +41,15 @@
     AC_CHECK_LIB(sun, socket,[SOCKET_FOUND=yes;SOCKLIBS="-lsun"])
   fi
 fi
+])
 dnl then gethostname
 GHN_FOUND=no
+AC_CHECK_FUNC(gethostname, [GHN_FOUND=yes], [
 AC_CHECK_LIB(nsl,gethostname,[GHN_FOUND=yes;SOCKLIBS="$SOCKLIBS -lnsl"])
 if test ${GHN_FOUND} = no; then
   AC_CHECK_LIB(nsl,gethostname,[GHN_FOUND=yes;SOCKLIBS="$SOCKLIBS -lsun"])
 fi
-
+])
 AC_SUBST(SOCKLIBS)
 
 dnl Checks for header files.
